libraries/lib-datahandler/src/main/thrift/moderation.thrift

```diff
@@ -55,6 +56,7 @@ struct ModerationRequest {
     17: optional string reviewer,
     18: required bool requestDocumentDelete,
     19: optional string requestingUserDepartment;
+    40: optional ComponentType componentType; // only relevant if the request is about components or releases
```

Comment:
I'd prefer to refrain from extending moderation requests with field that is specific for some document types. How about making sure that for component and release moderation requests, the component type is included in both `documentAdditions` and `documentDeletions` (thus indicating no change if not specifically requested) and using some utility method for extracting this from `documentAdditions` for displaying in the table? The utility method would default to empty for other moderation document types. Do you think that would work?

I think by going this route, you can even skip the db migration altogether. Moderation requests are relatively short-lived. I don't think it would be a big problem if you displayed the component type only for new moderation requests.

Reply:
Yes, I agree that adding the `componentType` is not beautiful. But I am also not happy with your suggestion. First, `...Additions` and `...Deletions` should be about be about the proposed changes. I wouldn't like using them to transport some extra information. And second, I do not think this would work for releases as the `ComponentType` is not included in the `Release` class.

Reply:
You're right. This won't work. I don't think the usage of additions and deletions would be a problem, but the `Release` not having a `ComponentType` is definitely one.
Alright, then. Let's leave it as it is.